### ISSUE TITLE
Test on DT4D animal meshes.

### ISSUE THREAD

**Issue by Kenkenzaii** [alekseizhuravlev/denoising-functional-maps#2]
Dear author,​​

​​Great work on your project! I tried testing the results on my own mesh data using the DT4D animal dataset. However, I encountered an issue where the matching results were completely scrambled. Did I miss any important steps or preprocessing requirements?​

**Comment by alekseizhuravlev** [comment]
Hi, thanks for interest in our work!

Could you please share the mesh files on which you tested and ideally the matching results that you got?

**Comment by Kenkenzaii** [comment]
Thank you so much for your help! I really appreciate your support.
Here are my visualization results, along with the mesh I used for testing:[https://drive.google.com/file/d/1QsDCt7Btar7gPBfYQQhDCeRVVohsMYU0/view?usp=sharing](url)

<img width="1509" height="646" alt="Image" src="https://github.com/user-attachments/assets/e8525350-7333-4365-afe6-17cb444717a7" />

**Comment by alekseizhuravlev** [comment]
So I experimented with the bear shape and got the same results as you. Looks like it's a failure case :)

I think the reasons are:
- The training dataset for the animal model consisted of 32 original SMAL shapes, each in 2000 poses. The topological structure wasn't very diverse since there were so few source shapes...
- The face / ears / tail of the bear shape are quite different from training shapes, which seems to confuse the model.

Maybe the performance on other meshes from DT4D-Animal will be better, would you mind sharing them?

**Comment by Kenkenzaii** [comment]
Thank you very much for your time and detailed explanation. Really appreciate it !
I also tested another shape from the DT4D animals dataset, which is very similar to the cow_01 mesh in the test set. It also produced unsatisfactory results. [https://drive.google.com/file/d/1Qj7cqwO5TW5g4zndyit-04LowR5iRIbz/v

<img width="1678" height="922" alt="Image" src="https://github.com/user-attachments/assets/4dbffa90-f74f-42a6-965f-697563ce1e62" />

iew?usp=sharing](url)

**Comment by alekseizhuravlev** [comment]
Ok, now I see what was happening.

In our pipeline, the model first generates N = 128 candidate point maps, and then selects the best one (Sec. 4.1.1 and Supp. B.3).
There was a bug in visualization of template-wise maps, where the **first** generated map was visualized instead of the **best** one :)

I fixed the bug now, here are the updated results. The models works for the bull mesh, but fails on the bear mesh.

Thanks for pointing this out!

| **Bull - Template** | **Bear - Template** | **Bull - Bear** |
|---------------------|---------------------|-----------------|
| <img width="999" height="373" alt="Bull - Template" src="https://github.com/user-attachments/assets/7c8e2acd-56eb-4833-a399-eda306dd7fb0" /> | <img width="1039" height="348" alt="Bear - Template" src="https://github.com/user-attachments/assets/d80a11cd-65fc-4edf-ab8c-358ef1dffd7a" /> | <img width="1151" height="523" alt="Bull - Bear" src="https://github.com/user-attachments/assets/74eb0392-9c6c-4b4e-8645-a364ef4aab51" /> |
